File: JIWON/012 주식가격.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> prices) {
    vector<int> answer;
    int count = 0;
    
    for(int i=0; i<prices.size(); i++){
        for(int j=i+1;j<prices.size();j++){
            if(prices[i] <= prices[j]){
                count++;
            } else{
                count++;
                break;
            }
        }
        
        answer.push_back(count);
        
        count = 0;
    }
    
    return answer;
}


//////////2020.07.31//////////

#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> prices) {
    vector<int> answer;
    int count = 0;
    
    for(int i=0;i<prices.size();i++){
        for(int j=i+1;j<prices.size();j++){
            count++;
            if(prices[i] > prices[j]) break;
        }
        answer.push_back(count);
        count = 0;
    }
    
    return answer;
}

File: lesunny/012_주식가격.java
class Solution {
    public int compare(int idx, int length, int[] prices){
        int cnt=0;
        for(int i=idx+1;i<length;i++){
            cnt++;
            if(prices[idx]>prices[i]) break;
        }
        return cnt;
    }
    public int[] solution(int[] prices) {
        int[] answer = {};
        int length = prices.length;
        answer = new int[length];
        for(int i=0;i<length;i++){
            answer[i] = compare(i, length, prices);
        }
        
        return answer;
    }
}
